src/useValidationObserver/useValidationObserver.spec.ts

```diff
@@ -0,0 +1,131 @@
+import { mount } from '@vue/test-utils'
+import { describe, it, expect, vi } from 'vitest'
+import { ref } from 'vue'
+import { useValidation } from '@/useValidation'
+import { useValidationObserver } from '@/useValidationObserver'
+import { timeout } from '@/utilities/tests'
+
+describe('useValidationObserver', () => {
+  it('should observe a single useValidation', async () => {
+    const validationRule = vi.fn().mockResolvedValue(true)
+    const wrapper = mount({
+      setup() {
+        const { validate } = useValidationObserver()
+        useValidation(ref(0), validationRule)
+        return { validate }
+      },
+    })
+
+    const result = await wrapper.vm.validate()
+
+    expect(result).toBe(true)
+    expect(validationRule).toHaveBeenCalled()
+  })
+
+  it('should observe multiple useValidations', async () => {
+    const validationRule1 = vi.fn().mockResolvedValue(true)
+    const validationRule2 = vi.fn().mockResolvedValue(true)
+    const wrapper = mount({
+      setup() {
+        const { validate } = useValidationObserver()
+        useValidation(ref(0), validationRule1)
+        useValidation(ref(0), validationRule2)
```

Comment:
it doesn't matter but for these tests where you're not changing the value you can just pass a static value. It doesn't have to be a ref.